app/collections/components/AdminTitle.tsx

```diff
@@ -1,54 +1,64 @@
-import { useMutation } from "blitz"
+import { useMutation, validateZodSchema } from "blitz"
 import { Field, Form, Formik } from "formik"
 import toast from "react-hot-toast"
 import changeTitle from "../mutations/changeTitle"
+import { useFormik } from "formik"
+import { z } from "zod"
 
 const AdminTitle = ({ collection, refetchFn, isAdmin }) => {
   const [changeTitleMutation] = useMutation(changeTitle)
 
+  const formik = useFormik({
+    initialValues: {
+      title: collection.title,
+    },
+    validate: validateZodSchema(
+      z.object({
+        title: z.string().min(1).max(300),
+      })
+    ),
+    onSubmit: async (values) => {
+      toast.promise(
+        changeTitleMutation({
+          id: collection.id,
+          title: values.title,
+        }),
+        {
+          loading: "Updating title...",
+          success: () => {
+            refetchFn()
+            return "Updated title"
+          },
+          error: "Failed to update title",
+        }
+      )
+    },
+  })
+
   return (
     <>
       {(collection.title === null || !collection.public || collection.upgraded) && isAdmin ? (
         <>
-          <Formik
-            initialValues={{
-              title: collection.title,
-            }}
-            onSubmit={() => {}}
+          <form
+            className="module divide-y divide-gray-100 border-0 border-gray-100 bg-white pb-2 dark:divide-gray-600 dark:border-gray-600 dark:bg-gray-900"
+            onSubmit={formik.handleSubmit}
           >
-            <Form
-              className="my-4 w-full"
-              onBlur={(values) => {
-                if (collection.title != values.target.defaultValue) {
-                  toast.promise(
-                    changeTitleMutation({
-                      id: collection.id,
-                      title: values.target.defaultValue,
-                    }),
-                    {
-                      loading: "Updating title...",
-                      success: () => {
-                        refetchFn()
-                        return "Updated title"
-                      },
-                      error: "Failed to update title",
-                    }
-                  )
-                }
-              }}
+            <label htmlFor="title" className="sr-only">
+              title
+            </label>
+            <textarea
+              rows={1}
+              id="title"
+              className="w-full select-none overflow-auto border-0 bg-white text-center text-3xl focus:ring-0 dark:bg-gray-900 md:text-5xl lg:text-6xl "
+              {...formik.getFieldProps("title")}
+            />
```

Comment:
Similarly from the above comment, it would be something like this?


```suggestion
            />
            {formik.errors.title && (
              <div className="border-none text-center text-sm font-medium text-gray-700 dark:text-gray-100">
                {"The title should be 1 to 300 characters long."}
              </div>
            )}
```

Reply:
Great suggestions! Updated.